tmail-backend/mailets/src/test/scala/com/linagora/tmail/mailets/EnforceRateLimitingPlanTest.scala

```diff
@@ -0,0 +1,596 @@
+        softly.assertThat(sentMails.get(0).getState).isEqualTo("error")
+        softly.assertThat(sentMails.get(0).getRecipients).containsExactlyInAnyOrder(USER1.asMailAddress())
+      })
+    }
+
+    @Test
+    def shouldRateLimitedWhenAllRecipientsExceeded(): Unit = {
+      val mail1: Mail = FakeMail.builder()
+        .name("mail")
+        .sender("sender1@domain.tld")
+        .recipients(USER1.asString(), USER2.asString())
+        .state("transport")
+        .build()
+
+      val mail2: Mail = FakeMail.builder()
+        .name("mail")
+        .sender("sender1@domain.tld")
+        .recipients(USER1.asString(), USER2.asString())
+        .state("transport")
+        .build()
+
+      mailet.service(mail1)
+      mailet.service(mail2)
+
+      assertSoftly(softly => {
+        softly.assertThat(mail1.getState).isEqualTo("transport")
+        softly.assertThat(mail2.getState).isEqualTo("error")
+      })
+    }
+
+    @Test
+    def shouldNotAppliedWhenOperationLimitationNotFound(): Unit = {
+      val rateLimitingPlan: RateLimitingPlan = SMono.fromPublisher(rateLimitationPlanRepository.create(RateLimitingPlanCreateRequest(
+        name = "LimitPlan",
+        operationLimitations = OperationLimitationsType.liftOrThrow(Seq(
+          TransitLimitations(Seq(RateLimitation(
+            name = "transit limit 1",
+            period = Duration.ofSeconds(100),
+            limits = LimitTypes.from(Map(("count", 1)))))))))))
+        .block()
+
+      val username: Username = Username.of("user3@domain.tld")
+      SMono.fromPublisher(rateLimitingPlanUserRepository.applyPlan(username, rateLimitingPlan.id)).block()
+      mailet = testee(mailetConfig.build())
+
+      val mail1: Mail = FakeMail.builder()
+        .name("mail")
+        .sender("sender1@domain.tld")
+        .recipients(username.asString())
+        .state("transport")
+        .build()
+
+      val mail2: Mail = FakeMail.builder()
+        .name("mail2")
+        .sender("sender1@domain.tld")
+        .recipients(username.asString())
+        .state("transport")
+        .build()
+
+      mailet.service(mail1)
+      mailet.service(mail2)
+
+      assertSoftly(softly => {
+        softly.assertThat(mail1.getState).isEqualTo("transport")
+        softly.assertThat(mail2.getState).isEqualTo("transport")
+      })
+    }
+  }
+
+  @Nested
+  class Configuration {
+    @Test
+    def shouldFailWhenNoOperationLimitation(): Unit = {
+      assertThatThrownBy(() => testee(FakeMailetConfig.builder()
+        .mailetName("EnforceRateLimitingPlan")
+        .build()))
+        .isInstanceOf(classOf[IllegalArgumentException])
+    }
+    @Test
+    def shouldFailWhenInvalidOperationLimitation(): Unit = {
+      assertThatThrownBy(() => testee(FakeMailetConfig.builder()
+        .mailetName("EnforceRateLimitingPlan")
+        .setProperty("operationLimitation","badOperation")
+        .build()))
+        .isInstanceOf(classOf[IllegalArgumentException])
+    }
+
+  }
+}
```

Comment:
I understood that relay and transit share the same sender logic, but shouldn't we have at least one test with relay to check quickly it has the intended behavior?